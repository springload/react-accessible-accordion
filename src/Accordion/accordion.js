// @flow

import React, { Component } from 'react';
import type { Node } from 'react';
import { isArraysEqualShallow } from '../utils';

type AccordionProps = {
    accordion: boolean,
    children: Node,
    activeItems: Array<string | number>,
    className: string,
    onChange: Function,
};

type AccordionState = {
    activeItems: Array<string | number>,
};

class Accordion extends Component<AccordionProps, AccordionState> {
    static defaultProps = {
        accordion: true,
        onChange: () => {},
        className: 'accordion',
        activeItems: [],
    };

    state = {
        activeItems: this.preExpandedItems(),
    };

    componentWillReceiveProps(
        nextProps: AccordionProps,
    ) {
        if (
            !isArraysEqualShallow(nextProps.activeItems, this.state.activeItems)
        ) {
            let newActiveItems;
            if (nextProps.accordion) {
                newActiveItems = nextProps.activeItems.length
                    ? [nextProps.activeItems[0]]
                    : [];
            } else {
                newActiveItems = nextProps.activeItems.slice();
            }
            this.setState({
                activeItems: newActiveItems,
            });

            nextProps.onChange(
                nextProps.accordion ? newActiveItems[0] : newActiveItems,
            );
        }
    }

    preExpandedItems() {
        let activeItems = [];
        React.Children.map(this.props.children, (item, index) => {
            if (item.props.expanded) {
                if (this.props.accordion) {
                    if (activeItems.length === 0)
                        activeItems.push(item.props.customKey || index);
                } else {
                    activeItems.push(item.props.customKey || index);
                }
            }
        });
        if (activeItems.length === 0 && this.props.activeItems.length !== 0) {
            activeItems = this.props.accordion
                ? [this.props.activeItems[0]]
                : this.props.activeItems.slice();
        }

        return activeItems;
    }

    handleClick(key: number | string) {
        let activeItems = this.state.activeItems;
        if (this.props.accordion) {
            activeItems = activeItems[0] === key ? [] : [key];
        } else {
            activeItems = [...activeItems];
            const index = activeItems.indexOf(key);
            const isActive = index > -1;
            if (isActive) {
                // remove active state
                activeItems.splice(index, 1);
            } else {
                activeItems.push(key);
            }
        }
        this.setState({
            activeItems,
        });

        this.props.onChange(
            this.props.accordion ? activeItems[0] : activeItems,
        );
    }

    renderItems() {
        const { accordion, children } = this.props;

        return React.Children.map(children, (item, index) => {
            const key = item.props.customKey || index;
            const expanded =
                this.state.activeItems.indexOf(key) !== -1 &&
                !item.props.disabled;

            return React.cloneElement(item, {
                disabled: item.props.disabled,
                accordion,
                expanded,
                key: `accordion__item-${key}`,
                onClick: this.handleClick.bind(this, key),
            });
        });
    }

    renderItems = this.renderItems.bind(this);

    render() {
        const { className, accordion } = this.props;
        return (
            <div role={accordion ? 'tablist' : null} className={className}>
                {this.renderItems()}
            </div>
        );
    }
}

export default Accordion;
