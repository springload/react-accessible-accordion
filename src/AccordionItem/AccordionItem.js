// @flow

import React, { Component } from 'react';
import type { ElementProps } from 'react';

import classNames from 'classnames';
import AccordionContainer from '../AccordionContainer/AccordionContainer';

type AccordionItemProps = ElementProps<'div'> & {
    uuid: string | number,
    hideBodyClassName: ?string,
    disabled: ?boolean,
    expanded: ?boolean,
    accordionStore: AccordionContainer,
};

class AccordionItem extends Component<AccordionItemProps, *> {
    componentDidMount() {
        const { uuid, accordionStore, disabled } = this.props;

        accordionStore.addItem({
            uuid,
            expanded: this.props.expanded || false,
            disabled,
        });
    }

    componentWillUnmount() {
        this.props.accordionStore.removeItem(this.props.uuid);
    }

    // This is here so that the user can dynamically set the 'expanded' state using the 'expanded' prop.
    componentDidUpdate(prevProps: AccordionItemProps) {
        const { uuid, expanded, accordionStore } = this.props;
        if (expanded !== prevProps.expanded) {
            accordionStore.setExpanded(uuid, expanded);
        }
    }

    render() {
        const {
            uuid,
            className,
            hideBodyClassName,
            accordionStore,
            disabled,
            expanded,
            ...rest
        } = this.props;

        const currentItem = accordionStore.state.items.find(
            item => item.uuid === uuid,
        );

        if (!currentItem) {
            return null;
        }

        return (
            <div
                className={classNames(className, {
                    [hideBodyClassName]:
                        !currentItem.expanded && hideBodyClassName,
                })}
                {...rest}
            />
        );
    }
}

export default AccordionItem;
