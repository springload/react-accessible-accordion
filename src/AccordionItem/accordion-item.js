// @flow

import React, { Component } from 'react';
import type { Node } from 'react';
import { inject, observer } from 'mobx-react';

import classNames from 'classnames';

type AccordionItemProps = {
    items: Array<Object>,
    children: Node,
    className: string,
    hideBodyClassName: string,
    itemkey: string | number,
    accordionStore: Object,
};

class AccordionItem extends Component<AccordionItemProps, *> {
    static defaultProps = {
        className: 'accordion__item',
        hideBodyClassName: '',
    };

    renderChildren() {
        const { children, itemkey } = this.props;

        return React.Children.map(children, item =>
            React.cloneElement(item, {
                itemkey,
            }),
        );
    }

    render() {
        const { className, hideBodyClassName } = this.props;
        const itemProperties = this.props.accordionStore.items.find(
            item => item.itemkey === this.props.itemkey,
        );

        if (!itemProperties) return null;
        const { expanded } = itemProperties;

        const itemClassName = classNames(className, {
            [hideBodyClassName]: !expanded && hideBodyClassName,
        });

        return <div className={itemClassName}>{this.renderChildren()}</div>;
    }
}

export default inject('accordionStore')(observer(AccordionItem));
