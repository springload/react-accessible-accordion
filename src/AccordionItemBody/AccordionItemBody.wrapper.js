// @flow

import React, { Component } from 'react';
import type { ElementProps } from 'react';

import { Subscribe } from 'unstated';
import AccordionContainer from '../AccordionContainer/AccordionContainer';
import ItemContainer from '../ItemContainer/ItemContainer';
import AccordionItemBody from './AccordionItemBody';

type AccordionItemBodyWrapperProps = ElementProps<'div'> & {
    hideBodyClassName: string,
};

class AccordionItemBodyWrapper extends Component<
    AccordionItemBodyWrapperProps,
> {
    static defaultProps = {
        className: 'accordion__body',
        hideBodyClassName: 'accordion__body--hidden',
    };

    renderItemBody = (
        accordionStore: AccordionContainer,
        itemStore: ItemContainer,
    ) => {
        const { uuid } = itemStore.state;
        const { items, accordion } = accordionStore.state;
        const item = items.find(stateItem => stateItem.uuid === uuid);
        return (
            <AccordionItemBody
                {...this.props}
                {...item}
                accordion={accordion}
            />
        );
    };

    render() {
        return (
            <Subscribe to={[AccordionContainer, ItemContainer]}>
                {this.renderItemBody}
            </Subscribe>
        );
    }
}

export default AccordionItemBodyWrapper;
