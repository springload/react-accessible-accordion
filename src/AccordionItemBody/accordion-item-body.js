import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const defaultProps = {
    id: '',
    expanded: false,
    className: 'accordion__body',
    hideBodyClassName: 'accordion__body--hidden',
    role: '',
};

const propTypes = {
    id: PropTypes.string,
    expanded: PropTypes.bool,
    children: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.arrayOf(PropTypes.element),
    ]).isRequired,
    className: PropTypes.string,
    hideBodyClassName: PropTypes.string,
    role: PropTypes.string,
};

const AccordionItemBody = ({ id, expanded, children, className, hideBodyClassName, role }) => {
    const bodyClass = classNames(
        className,
        {
            [hideBodyClassName]: !expanded,
        },
    );
    const ariaHidden = !expanded;
    return (
        <div
            id={id}
            className={bodyClass}
            aria-hidden={ariaHidden}
            aria-labelledby={id.replace('accordion__body-', 'accordion__title-')}
            role={role}
        >
            {children}
        </div>
    );
};

AccordionItemBody.propTypes = propTypes;
AccordionItemBody.defaultProps = defaultProps;
// We need this to be able to assign correct params to element.
// Minifiers modify component name
AccordionItemBody.accordionElementName = 'AccordionItemBody';

export default AccordionItemBody;
