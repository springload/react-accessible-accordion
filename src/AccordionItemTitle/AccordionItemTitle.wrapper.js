// @flow

import React, { Component } from 'react';
import type { ElementProps } from 'react';

import { Subscribe } from 'unstated';
import AccordionContainer from '../AccordionContainer/AccordionContainer';
import ItemContainer from '../ItemContainer/ItemContainer';
import AccordionItemTitle from './AccordionItemTitle';

type AccordionItemTitleWrapperProps = ElementProps<'div'> & {
    hideBodyClassName: string,
};

class AccordionItemTitleWrapper extends Component<
    AccordionItemTitleWrapperProps,
> {
    static defaultProps = {
        className: 'accordion__title',
        hideBodyClassName: '',
    };

    renderItemTitle = (
        accordionStore: AccordionContainer,
        itemStore: ItemContainer,
    ) => {
        const { uuid } = itemStore.state;
        const { items, accordion } = accordionStore.state;
        const item = items.find(stateItem => stateItem.uuid === uuid);

        return (
            <AccordionItemTitle
                {...this.props}
                {...item}
                setExpanded={accordionStore.setExpanded}
                accordion={accordion}
            />
        );
    };

    render() {
        return (
            <Subscribe to={[AccordionContainer, ItemContainer]}>
                {this.renderItemTitle}
            </Subscribe>
        );
    }
}

export default AccordionItemTitleWrapper;
