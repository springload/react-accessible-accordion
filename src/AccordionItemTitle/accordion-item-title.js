// @flow

import React, { Component } from 'react';
import type { Node } from 'react';
import classNames from 'classnames';
import { inject, observer } from 'mobx-react';

type AccordionItemTitleProps = {
    children: Node,
    className: string,
    hideBodyClassName: string,
    accordionStore: {
        items: Array<Object>,
        accordion: boolean,
        onChange: Function,
    },
    itemkey: string | number,
};

type AccordionItemTitleState = {};

export class AccordionItemTitle extends Component<
    AccordionItemTitleProps,
    AccordionItemTitleState,
> {
    static accordionElementName = 'AccordionItemTitle';

    static defaultProps = {
        className: 'accordion__title',
        hideBodyClassName: '',
    };

    handleClick() {
        const { itemkey } = this.props;
        const { accordion, onChange } = this.props.accordionStore;
        let { items } = this.props.accordionStore;
        const foundItem = items.find(item => item.itemkey === itemkey);
        if (!foundItem) return;

        if (accordion) {
            const newValue = !foundItem.expanded;
            items = items.map(item => {
                const resetItem = item;
                resetItem.expanded = false;
                return resetItem;
            });
            foundItem.expanded = newValue;
        } else {
            foundItem.expanded = !foundItem.expanded;
        }

        if (accordion) {
            onChange(foundItem.itemkey);
        } else {
            const filteredItems = items.filter(item => item.expanded === true);
            const itemkeys = filteredItems.map(item => item.itemkey);
            onChange(itemkeys);
        }
    }

    handleClick = this.handleClick.bind(this);

    handleKeyPress(evt: SyntheticInputEvent<HTMLButtonElement>) {
        if (evt.charCode === 13 || evt.charCode === 32) {
            this.handleClick();
        }
    }

    handleKeyPress = this.handleKeyPress.bind(this);

    render() {
        const { items, accordion } = this.props.accordionStore;
        const { itemkey, children, className, hideBodyClassName } = this.props;
        const foundItem = items.find(item => item.itemkey === itemkey);
        if (!foundItem) return null;

        const { itemUuid, expanded } = foundItem;

        const id = `accordion__title-${itemUuid}`;
        const ariaControls = `accordion__body-${itemUuid}`;
        const role = accordion ? 'tab' : 'button';
        const titleClassName = classNames(className, {
            [hideBodyClassName]: hideBodyClassName && !expanded,
        });

        if (role === 'tab') {
            return (
                <div
                    id={id}
                    aria-selected={expanded}
                    aria-controls={ariaControls}
                    className={titleClassName}
                    onClick={this.handleClick}
                    role={role}
                    tabIndex="0" // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
                    onKeyPress={this.handleKeyPress}
                >
                    {children}
                </div>
            );
        }
        return (
            <div
                id={id}
                aria-expanded={expanded}
                aria-controls={ariaControls}
                className={titleClassName}
                onClick={this.handleClick}
                role={role}
                tabIndex="0" // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
                onKeyPress={this.handleKeyPress}
            >
                {children}
            </div>
        );
    }
}

export default inject('accordionStore')(observer(AccordionItemTitle));
