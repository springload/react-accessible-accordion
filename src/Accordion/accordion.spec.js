// @flow

import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Accordion from './accordion';
import AccordionItem from '../AccordionItem/accordion-item';
import AccordionItemTitle from '../AccordionItemTitle/accordion-item-title';
import AccordionItemBody from '../AccordionItemBody/accordion-item-body';

jest.mock('../AccordionItem/accordion-item', () => 'div');
jest.mock('../AccordionItemTitle/accordion-item-title', () => 'div');
jest.mock('../AccordionItemBody/accordion-item-body', () => 'div');

describe('Accordion', () => {
    it('renders correctly with min params', () => {
        const tree = renderer
            .create(
                <Accordion>
                    <AccordionItem>Fake child</AccordionItem>
                    <AccordionItem>Fake child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with accordion false', () => {
        const tree = renderer
            .create(
                <Accordion accordion={false}>
                    <AccordionItem>Fake child</AccordionItem>
                    <AccordionItem>Fake child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('different className', () => {
        const tree = renderer
            .create(
                <Accordion accordion={false} className="testCSSClass">
                    <AccordionItem>Fake Child</AccordionItem>
                    <AccordionItem>Fake Child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handleClick function accordion true', () => {
        const wrapper = renderer.create(
            <Accordion>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        wrapper.getInstance().handleClick(1);
        expect(wrapper.getInstance().state.activeItems).toEqual([1]);
        expect(wrapper).toMatchSnapshot();

        wrapper.getInstance().handleClick(1);
        expect(wrapper.getInstance().state.activeItems).toEqual([]);
        expect(wrapper).toMatchSnapshot();
    });

    it('handleClick function accordion false', () => {
        const wrapper = renderer.create(
            <Accordion accordion={false}>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        wrapper.getInstance().handleClick(1);
        expect(wrapper.getInstance().state.activeItems).toEqual([1]);
        expect(wrapper).toMatchSnapshot();

        wrapper.getInstance().handleClick(1);
        expect(wrapper.getInstance().state.activeItems).toEqual([]);
        expect(wrapper).toMatchSnapshot();

        wrapper.getInstance().handleClick(0);
        wrapper.getInstance().handleClick(1);
        expect(wrapper.getInstance().state.activeItems).toEqual([0, 1]);
        expect(wrapper).toMatchSnapshot();
    });

    it('handles disabled children', () => {
        const wrapper = renderer.create(
            <Accordion accordion={false}>
                <AccordionItem disabled={true}>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        wrapper.getInstance().handleClick(0);
        expect(wrapper).toMatchSnapshot();
    });

    it('pre expanded accordion', () => {
        const tree = renderer
            .create(
                <Accordion>
                    <AccordionItem expanded={true}>Fake Child</AccordionItem>
                    <AccordionItem>Fake Child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('works with multiple pre expanded accordion. Extra expands are just ignored.', () => {
        const tree = renderer
            .create(
                <Accordion>
                    <AccordionItem expanded={true}>Fake Child</AccordionItem>
                    <AccordionItem expanded={true}>Fake Child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('pre expanded accordion when accordion is false', () => {
        const tree = renderer
            .create(
                <Accordion accordion={false}>
                    <AccordionItem expanded={true}>Fake Child</AccordionItem>
                    <AccordionItem expanded={true}>Fake Child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('pre expand accordion via accordion props', () => {
        const tree = renderer
            .create(
                <Accordion activeItems={[0]}>
                    <AccordionItem>Fake Child</AccordionItem>
                    <AccordionItem>Fake Child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('pre expand accordion via accordion props vs accordion item props. Expanded only second item.', () => {
        const tree = renderer
            .create(
                <Accordion activeItems={[0]}>
                    <AccordionItem>Fake Child</AccordionItem>
                    <AccordionItem expanded={true}>Fake Child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('pre expand multiple accordions via accordion props', () => {
        const tree = renderer
            .create(
                <Accordion accordion={false} activeItems={[0, 2]}>
                    <AccordionItem>Fake Child</AccordionItem>
                    <AccordionItem>Fake Child</AccordionItem>
                    <AccordionItem>Fake Child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('pre expand accordion via accordion props with custom key', () => {
        const tree = renderer
            .create(
                <Accordion activeItems={['custom']}>
                    <AccordionItem>Fake Child</AccordionItem>
                    <AccordionItem customKey="custom">Fake Child</AccordionItem>
                </Accordion>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('expand accordion via accordion props dynamicly', () => {
        const wrapper = renderer.create(
            <Accordion activeItems={[0]}>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        wrapper.update(
            <Accordion activeItems={[1]}>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        expect(wrapper.getInstance().state.activeItems).toEqual([1]);
        expect(wrapper).toMatchSnapshot();
    });

    it('expand multiple accordions via accordion props props dynamicly', () => {
        const wrapper = renderer.create(
            <Accordion accordion={false} activeItems={[0, 2]}>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        wrapper.update(
            <Accordion accordion={false} activeItems={[1, 2]}>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        expect(wrapper.getInstance().state.activeItems).toEqual([1, 2]);
        expect(wrapper).toMatchSnapshot();
    });

    it(`expand multiple accordions via accordion props props dynamicly with default
        expanded on accordion items`, () => {
        const wrapper = renderer.create(
            <Accordion accordion={false}>
                <AccordionItem expanded={true}>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        wrapper.update(
            <Accordion accordion={false} activeItems={[1, 2]}>
                <AccordionItem expanded={true}>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        expect(wrapper.getInstance().state.activeItems).toEqual([1, 2]);
        expect(wrapper).toMatchSnapshot();
    });

    it('close accordions via accordion props props dynamicly', () => {
        const wrapper = renderer.create(
            <Accordion activeItems={[1]}>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        wrapper.update(
            <Accordion activeItems={[]}>
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        expect(wrapper.getInstance().state.activeItems).toEqual([]);
        expect(wrapper).toMatchSnapshot();
    });

    it('different className with the same activeItems prop', () => {
        const wrapper = renderer.create(
            <Accordion activeItems={[1]} className="test-1">
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        wrapper.update(
            <Accordion activeItems={[1]} className="test-2">
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('different className with the same activeItems prop', () => {
        const wrapper = renderer.create(
            <Accordion activeItems={[1]} className="test-1">
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        wrapper.update(
            <Accordion activeItems={[1]} className="test-2">
                <AccordionItem>Fake Child</AccordionItem>
                <AccordionItem>Fake Child</AccordionItem>
            </Accordion>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('supports controlled component inside accordion', () => {
        class App extends Component<*, { value: string }> {
            constructor(props) {
                super(props);

                this.state = {
                    value: '',
                };
            }

            handleChange = evt => {
                this.setState({
                    value: evt.target.value,
                });
            };

            render() {
                return (
                    <Accordion activeItems={[1]} id="accordion">
                        <AccordionItem id="accordion-item--1">
                            <AccordionItemTitle>
                                {`Title One`}
                            </AccordionItemTitle>
                            <AccordionItemBody>
                                <input
                                    id="controlled-input--1"
                                    onChange={this.handleChange}
                                    value={this.state.value}
                                />
                            </AccordionItemBody>
                        </AccordionItem>
                        <AccordionItem>
                            <AccordionItemTitle>
                                {`Title Two`}
                            </AccordionItemTitle>
                            <AccordionItemBody>
                                {`Body Two`}
                            </AccordionItemBody>
                        </AccordionItem>
                    </Accordion>
                );
            }
        }
        const wrapper = mount(<App />);
        const accordion = wrapper.find(Accordion);
        const input = wrapper.find('#controlled-input--1');
        const itemOne = wrapper.find('#accordion-item--1');

        expect(accordion.instance().state.activeItems).toEqual([1]);
        itemOne.simulate('click');
        expect(accordion.instance().state.activeItems).toEqual([0]);
        input.simulate('change', { value: 'foo' });
        expect(accordion.instance().state.activeItems).toEqual([0]);
    });
});
