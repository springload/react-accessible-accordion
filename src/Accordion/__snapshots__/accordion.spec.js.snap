// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion close accordions via accordion props props dynamicly 1`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion different className 1`] = `
<div
  className="testCSSClass"
  role={null}
>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion different className with the same activeItems prop 1`] = `
<div
  className="test-2"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion different className with the same activeItems prop 2`] = `
<div
  className="test-2"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion expand accordion via accordion props dynamicly 1`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion expand multiple accordions via accordion props props dynamicly 1`] = `
<div
  className="accordion"
  role={null}
>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion expand multiple accordions via accordion props props dynamicly with default
        expanded on accordion items 1`] = `
<div
  className="accordion"
  role={null}
>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion handleClick function accordion false 1`] = `
<div
  className="accordion"
  role={null}
>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion handleClick function accordion false 2`] = `
<div
  className="accordion"
  role={null}
>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion handleClick function accordion false 3`] = `
<div
  className="accordion"
  role={null}
>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion handleClick function accordion true 1`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion handleClick function accordion true 2`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion handles disabled children 1`] = `
<div
  className="accordion"
  role={null}
>
  <div
    accordion={false}
    disabled={true}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion pre expand accordion via accordion props 1`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion pre expand accordion via accordion props vs accordion item props. Expanded only second item. 1`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion pre expand accordion via accordion props with custom key 1`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    customKey="custom"
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion pre expand multiple accordions via accordion props 1`] = `
<div
  className="accordion"
  role={null}
>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion pre expanded accordion 1`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion pre expanded accordion when accordion is false 1`] = `
<div
  className="accordion"
  role={null}
>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;

exports[`Accordion renders correctly with accordion false 1`] = `
<div
  className="accordion"
  role={null}
>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake child
  </div>
  <div
    accordion={false}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake child
  </div>
</div>
`;

exports[`Accordion renders correctly with min params 1`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake child
  </div>
</div>
`;

exports[`Accordion works with multiple pre expanded accordion. Extra expands are just ignored. 1`] = `
<div
  className="accordion"
  role="tablist"
>
  <div
    accordion={true}
    disabled={undefined}
    expanded={true}
    onClick={[Function]}
  >
    Fake Child
  </div>
  <div
    accordion={true}
    disabled={undefined}
    expanded={false}
    onClick={[Function]}
  >
    Fake Child
  </div>
</div>
`;
